distributed/scheduler.py

```diff
@@ -3144,15 +3145,18 @@ def coerce_address(self, addr, resolve=True):
         """
         # XXX how many address-parsing routines do we have?
         if addr in self.aliases:
-            addr = self.aliases[addr]
+            if clear_cache:
+                del self.aliases[addr]
```

Comment:
Deleting the aliases should only happen during a remove_worker event.  Placing this code here confuses me.

Reply:
You are right, and this is actually redundant here.